library/psa_its_file.c

```diff
@@ -214,9 +214,12 @@ psa_status_t psa_its_set( psa_storage_uid_t uid,
     n = fwrite( &header, 1, sizeof( header ), stream );
     if( n != sizeof( header ) )
         goto exit;
-    n = fwrite( p_data, 1, data_length, stream );
-    if( n != data_length )
-        goto exit;
+    if( p_data != NULL )
```

Comment:
This should be `if( data_length == 0 )`. Checking for null silently writes nothing instead of faulting if `p_data` is null but `data_length != 0`.